### ISSUE TITLE
Creating images getting "Network Error" message.

### ISSUE THREAD

**Issue by pblee88** [harvester/harvester#525]
When creating Harvester images. After place the image URL and click create button. Getting "Network Error" message. I have check the host able to ping and curl to my own web server which contain image file. Any idea what is wrong on this?

**Comment by gitlawr** [comment]
Can you provide the following information:
 - Harvester version: 
 - Installation Mode: ISO or app-mode

Can you also check the requests' status in the browser console?

**Comment by pblee88** [comment]
* Harvester version:

Ans: v0.1.0

  * Installation Mode: ISO or app-mode

Ans: ISO

**Comment by pblee88** [comment]
Today I am getting another error message while creating images.

Get http://minio.harvester-system:9000/vm-images/?location=: dial tcp 10.43.187.44:9000: connect: connection refused
--

**Comment by gitlawr** [comment]
Can you ssh to the node and run `kubectl get po -n harvester-system  -l app=minio` to check?

Regarding the original issue, Did you check the API call status in the browser console?

**Comment by pblee88** [comment]
Here is the screenshot

Screenshot for first question.
![Screenshot for kubectl command](https://user-images.githubusercontent.com/30486081/110923060-6e41ca00-835b-11eb-8cc4-9d295017d9dc.png)

Screenshot for 2nd question.
![Screenshot for Network Error message](https://user-images.githubusercontent.com/30486081/110923064-700b8d80-835b-11eb-9ba8-b373fe762256.png)

**Comment by pblee88** [comment]
Attach is the screen shoot after re run the kubectl get po -n harvester-system -l app=minio command again.

![image](https://user-images.githubusercontent.com/30486081/111240006-56a56280-8635-11eb-81be-674fcc12a9e8.png)

**Comment by gitlawr** [comment]
1. For the cert issue, you can trust the CA cert in your system, or try to re-enable the certificate check in chrome, then proceed by ignoring the warnings.
![image](https://user-images.githubusercontent.com/5697937/111248088-2d400300-8644-11eb-81fa-762ee75c8eb8.png)

2. You can run `kubectl describe po minio-3 -n harvester-system` to check why it is in a pending state.

**Comment by pblee88** [comment]
I manage to fix the cert issue. Need to go to "Internet Properties"/"Content" tab/ and clock on clear SSL state. 

Regarding the issue 2nd issue. After run the kubectl describe po minio-3 -n harvester-system command. Below is the output.

Name:           minio-3
Namespace:      harvester-system
Priority:       0
Node:           <none>
Labels:         app=minio
                controller-revision-hash=minio-8b9c97c79
                release=harvester
                statefulset.kubernetes.io/pod-name=minio-3
Annotations:    checksum/config: 2ff72f90513284e65dbc8b6c69944b498363cbb19814f893f2ff0b03e776a8a8
                checksum/secrets: 0fe4d957694cc0beba0b2af79696d2c063dc3a4d869b769d2cffd805c333c4d9
Status:         Pending
IP:
IPs:            <none>
Controlled By:  StatefulSet/minio
Containers:
  minio:
    Image:      minio/minio:RELEASE.2020-11-19T23-48-16Z
    Port:       9000/TCP
    Host Port:  0/TCP
    Command:
      /bin/sh
      -ce
      /usr/bin/docker-entrypoint.sh minio -S /etc/minio/certs/ server  http://minio-{0...3}.minio-svc.harvester-system.svc.cluster.local/export
    Limits:
      cpu:     2
      memory:  4Gi
    Requests:
      cpu:     500m
      memory:  512Mi
    Environment:
      MINIO_ACCESS_KEY:  <set to the key 'accesskey' in secret 'minio'>  Optional: false
      MINIO_SECRET_KEY:  <set to the key 'secretkey' in secret 'minio'>  Optional: false
    Mounts:
      /export from export (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from minio-token-mjxkq (ro)
Conditions:
  Type           Status
  PodScheduled   False
Volumes:
  export:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  export-minio-3
    ReadOnly:   false
  minio-user:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  minio
    Optional:    false
  minio-token-mjxkq:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  minio-token-mjxkq
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason            Age   From               Message
  ----     ------            ----  ----               -------
  Warning  FailedScheduling  167m  default-scheduler  0/2 nodes are available: 2 pod has unbound immediate PersistentVolumeClaims.


Creating images still getting below error.
Get http://minio.harvester-system:9000/vm-images/?location=: dial tcp 10.43.187.44:9000: connect: connection refused

**Comment by gitlawr** [comment]
>pod has unbound immediate PersistentVolumeClaims.

You can run the following to check why PVCs are unbound
```
$ kubectl -n harvester-system get pvc
$ kubectl -n harvester-system describe pvc <pvc-name>
```

**Comment by pblee88** [comment]
Base on the command you provided. It look like my VM is running  out of resources. I am setting up another set of Rancher/Harvester on 2 bare metal server. My bare metal system have plenty of resource with 7TB local disk. But I am not understand why main server Rancher/Harvester installation only auto partition to 20GB disk space. Remaining disk space is not using. 2nd bare metal server configure to join existing cluster and able to utilize full 7TB disk space. Is this normal? Because I install the server on Linux KVM with 200GB disk space, and it utilize full 200GB disk space.

**Comment by pblee88** [comment]
Attach is the screen capture after installation. only 19GB partition created on /dev/sda1. 
![image](https://user-images.githubusercontent.com/30486081/111960349-9800ab00-8b2a-11eb-8092-5a2d3349bdf4.png)

**Comment by gitlawr** [comment]
Offline resize is supported with the following steps:
```
# scale down minio statefulset to 0
$ kubectl -n harvester-system scale sts minio --replicas=0
# Wait for all minio pods to be removed.

# patch the pvc size, the following 30Gi is customizable depending on the environment
$ kubectl -n harvester-system patch pvc export-minio-{0,1,2,3} -p '{"spec":{"resources":{"requests":{"storage":"30Gi"}}}}}'

# scale back the minio
$ kubectl -n harvester-system scale sts minio --replicas=0
```

We are planning to move out the minIO component so this limit is hopefully gone in the next release

**Comment by pblee88** [comment]
Sorry for late update. I am hitting some issue on the network. I manage to fix network issue by re setup the Rancher/Harvester. Now disk size problem still exist. OS only create 20GB partition on 7TB volume disk to host OS. I have try the given command to resize partition. i am getting the below error.

The server I am working on is compute node. Below is the screenshot of the error message.
![image](https://user-images.githubusercontent.com/30486081/113076079-e5eb7200-9200-11eb-96f4-fdd6b6d60e5a.png)

Below screenshot can see the OS only installed on 20GB partition.
![Uploading image.png…]()

**Comment by gitlawr** [comment]
If you are working on a compute node, you can add the specify kubeconfig path before running the above commands
e.g.,
```
sudo KUBECONFIG=/var/lib/rancher/k3s/agent/kubelet.kubeconfig kubectl get node
```

**Comment by w13915984028** [comment]
A known issue (fast download causes POD OOM killed) is solved after v1.0.0, close.
